PyInstaller/building/api.py

```diff
@@ -171,8 +177,44 @@ def assemble(self):
                 self.code_dict[entry[0]] = self.__get_code(entry[0], entry[1])
         # sort content alphabetically to support reproducible builds
         toc.sort()
+
+        # Remove leading parts of paths in code objects
+        self.code_dict = {
+            key: self._strip_paths_in_code(code)
+            for key, code in self.code_dict.items()
+        }
```

Comment:
Wow, this is interesting. I've never seen this before. I used to write `dict((key, value) for ...)`. But this is much nicer :-)